pyanaconda/modules/payloads/source/harddrive/initialization.py

```diff
@@ -0,0 +1,112 @@
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+import os.path
+
+from pyanaconda.core.storage import device_matches
+
+from pyanaconda.modules.common.errors.payload import SourceSetupError
+from pyanaconda.modules.common.task import Task
+from pyanaconda.modules.common.errors.storage import MountFilesystemError, DeviceSetupError
+
+from pyanaconda.payload.utils import mount, unmount, PayloadSetupError
+from pyanaconda.payload.image import verify_valid_installtree, find_first_iso_image
+
+from pyanaconda.anaconda_loggers import get_module_logger
+log = get_module_logger(__name__)
+
+__all__ = ["SetUpHardDriveSourceTask", "TearDownHardDriveSourceTask"]
+
+
+class TearDownHardDriveSourceTask(Task):
+    """Task to teardown installation source."""
+
+    def __init__(self, target_mount):
+        super().__init__()
+        self._device_mount = target_mount + "_device"
+        self._iso_mount = target_mount + "_iso"
+
+    @property
+    def name(self):
+        return "Tear down Hard drive installation source"
+
+    def run(self):
+        """Run live installation source un-setup."""
+        log.debug("Unmounting Hard drive installation source")
+        unmount(self._device_mount)
+        unmount(self._iso_mount)
+
+
+class SetUpHardDriveSourceTask(Task):
+    """Task to setup installation source."""
+
+    def __init__(self, target_mount, partition, directory):
+        # TODO need better name for target_mount
+        super().__init__()
+        self._device_mount = target_mount + "_device"
+        self._iso_mount = target_mount + "_iso"
+        self._partition = partition
+        self._directory = directory
+
+    @property
+    def name(self):
+        return "Set up Hard drive installation source"
+
+    def run(self):
+        """Run Hard drive installation source setup.
+
+        :return: Path to the install tree.
+        :rtype: str
+        """
+        log.debug("Setting up Hard drive source")
+
+        try:
+            matches = device_matches(self._partition)
+            if not matches:
+                raise SourceSetupError(
+                    "Device for Hard drive install {} does not exist".format(self._partition)
+                )
+
+            mount("/dev/" + matches[0], self._device_mount, "auto", "")
+
+        except (DeviceSetupError, MountFilesystemError, PayloadSetupError, SourceSetupError) as e:
+            log.error("Mount of partition device failed: %s", e)
+            raise SourceSetupError(str(e))
+
+        full_path_on_mounted_device = os.path.normpath(os.path.join(self._device_mount,
+                                                                    self._directory))
```

Comment:
Use this when it's merged:
https://github.com/rhinstaller/anaconda/pull/2426

Reply:
Doing a normpath of format for now...

Reply:
That won't help here. But you can ignore it I'll fix it in a separate PR if required.

Reply:
Normpath normalizes double slashes, so that should work.
```
full_path_on_mounted_device = os.path.normpath(
            "{}/{}".format(self._device_mount, self._directory)
        )
```